packages/flutter_tools/lib/src/migrate/migrate_update_locks.dart

```diff
@@ -0,0 +1,128 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+
+import '../base/file_system.dart';
+import '../base/logger.dart';
+import '../base/terminal.dart';
+import '../project.dart';
+import 'migrate_utils.dart';
+
+/// Checks if the project uses pubspec dependency locking and prompts if
+/// the pub upgrade should be run.
+Future<void> updatePubspecDependencies(
+    FlutterProject flutterProject,
+    MigrateUtils migrateUtils,
+    Logger logger,
+    Terminal terminal
+) async {
+  final File pubspecFile = flutterProject.directory.childFile('pubspec.yaml');
+  if (!pubspecFile.existsSync()) {
+    return;
+  }
+  if (!pubspecFile.readAsStringSync().contains('# THIS LINE IS AUTOGENERATED')) {
+    return;
+  }
+  logger.printStatus('\nDart dependency locking detected in pubspec.yaml.');
+  String selection = 'y';
+  try {
+    selection = await terminal.promptForCharInput(
+      <String>['y', 'n'],
+      logger: logger,
+      prompt: 'Do you want the tool to run `flutter pub upgrade --major-versions`? (y)es, (n)o',
+      defaultChoiceIndex: 1,
+    );
+  } on StateError catch(e) {
```

Comment:
Why do we catch `StateError` here?

Reply:
It is thrown if useTerminalUi is false, but since we now set it to true right before, I can just remove this try-catch block